src/eckit/parser/YAMLParser.cc

```diff
@@ -443,27 +479,30 @@ static Value toValue(const std::string& s) {
     if (s.length()) {
         // This is because checking regex is very slow
         switch (s[0]) {
+            case '-':
+            case '+':
             case '0':
 
-                if (octal.match(s)) {
-                    return Value(strtol(s.c_str(), 0, 0));
+                if (integer2.match(s)) {
+                    std::string str = ignoreChar(s, '_');
+                    if (s[1] == 'b') {  // 0b..., no sign
+                        str = str.substr(2);
+                    } else {  // +0b... or -0b..., signed
+                        str = str.substr(0, 1) + str.substr(3);
+                    }
+                    return Value(strtol(str.c_str(), 0, 2));
                 }
 
-                if (s.length() > 2 && s[1] == 'x' && hex.match(s)) {
-                    return Value(strtol(s.c_str(), 0, 0));
+                if (integer8.match(s)) {
+                    std::string str = ignoreChar(s, '_');
+                    return Value(strtol(str.c_str(), 0, 0));
```

Comment:
Should we not be being explicit about base 8 here, given we have already tested for it?

Reply:
Done.